pkg/storage/ingest/reader.go

```diff
@@ -90,6 +90,10 @@ func newPartitionReader(kafkaCfg KafkaConfig, partitionID int32, consumerGroup s
 }
 
 func (r *PartitionReader) start(ctx context.Context) (returnErr error) {
+	if r.kafkaCfg.DefaultPartitionsForAutocreatedTopics > 0 {
```

Comment:
I would also do it only when `AutoCreateTopicEnabled` is true. WDYT?

_If you agree, same comment applies to the writer._